### ISSUE TITLE
Iphone safair support webkitEnterFullscreen.

### ISSUE THREAD

**Issue by TianbinTobin** [sindresorhus/screenfull.js#221]
Iphone safair support webkitEnterFullscreen, not webkitRequestFullscreen

**Comment by DmytroShulzhenko** [comment]
Can you add webkitEnterFullscreen for supporting full screen mode for Iphones?

**Comment by justingolden21** [comment]
+1 to this, not working on iphone

**Comment by WINOFFRG** [comment]
+1 Much Needed, Currently have to work with `webkitEnterFullscreen` and `webkitExitFullscreen`

**Comment by chrisandrews7** [comment]
+1

**Comment by BePo65** [comment]
There seems to be a small bug in the screenfull code (lines 14 and 24):

For apple / safari it uses `webkitRequestFullscreen` (in compliance with mdn and CanIuse).

But according to the apple [documentation](https://developer.apple.com/documentation/webkitjs/htmlvideoelement/1633500-webkitenterfullscreen), safari not only requires the `webkit` prefix, but also uses a different wording: `webkitEnterFullscreen`.

**Comment by justingolden21** [comment]
If I (or someone else) make a PR for this, is there any chance it would be merged?

@ screenfull team

**Comment by aquinoWill** [comment]
+1

**Comment by BePo65** [comment]
I will prepare a pr for this; only testing on ios is a little bit difficult for me.

**Comment by baharev** [comment]
I have tested it on an iPhone with Safari 17.2 and based on my own experiments and on the [official docs](https://developer.apple.com/documentation/webkitjs/htmlvideoelement/1633500-webkitenterfullscreen): 

**`webkitEnterFullscreen` is only available for the `HTMLVideoElement`** as of Feb 9, 2024.

It makes little sense to me, since full screen is the default for the video element on the iPhone and you have to opt-out by specifying `playsinline` if you don't want that. The video element also has its own control to toggle the full screen mode, making `webkitEnterFullscreen` superfluous.

For other HTML elements, `webkitEnterFullscreen` seems to be absent on the iPhone. 

Long story short: Even if `webkitEnterFullscreen` is added to screenfull library, it won't make a difference on an iPhone.

**Comment by BePo65** [comment]
Perhaps I can help with a little [web page](https://bepo65.github.io/fullscreen-features/index.html) that shows the fullscreen features available on 'your' browser.

**Comment by baharev** [comment]
> Perhaps I can help with a little [web page](https://bepo65.github.io/fullscreen-features/index.html) that shows the fullscreen features available on 'your' browser.

I don't get the point you are trying to make but here is a screenshot made on that iPhone with Safari 17.2

![IMG_4950](https://github.com/sindresorhus/screenfull/assets/2605584/f206a626-7f9c-4d1b-942d-c65cb635b14c)

**Comment by BePo65** [comment]
The page just proves that iPhone does not support the Fullscreen API (besides for HTMLVideoElement) as you said before and thus a or is useless.

**Comment by cay-cospete** [comment]
> I have tested it on an iPhone with Safari 17.2 and based on my own experiments and on the [official docs](https://developer.apple.com/documentation/webkitjs/htmlvideoelement/1633500-webkitenterfullscreen):
> 
> **`webkitEnterFullscreen` is only available for the `HTMLVideoElement`** as of Feb 9, 2024.
> 
> It makes little sense to me, since full screen is the default for the video element on the iPhone and you have to opt-out by specifying `playsinline` if you don't want that. The video element also has its own control to toggle the full screen mode, making `webkitEnterFullscreen` superfluous.
> 
> For other HTML elements, `webkitEnterFullscreen` seems to be absent on the iPhone.
> 
> Long story short: Even if `webkitEnterFullscreen` is added to screenfull library, it won't make a difference on an iPhone.

Why is that `playsInline` attribute is correlated with fullscreen mode on Safari ? Also, why is that adding `webkitEnterFullscreen` in the fullscreen library wont make a difference ?

**Comment by BePo65** [comment]
>Also, why is that adding webkitEnterFullscreen in the fullscreen library wont make a difference ?

If you take a look at my [testpage](https://bepo65.github.io/fullscreen-features/index.html) with an iPhone (I don't have one available): what does it say for 'webkitEnterFullscreen'?

**Comment by cay-cospete** [comment]
> > Also, why is that adding webkitEnterFullscreen in the fullscreen library wont make a difference ?
> 
> If you take a look at my [testpage](https://bepo65.github.io/fullscreen-features/index.html) with an iPhone (I don't have one available): what does it say for 'webkitEnterFullscreen'?

It says unavailable for all but i am trying to understand why ?, when i tested on iphone 12 pro on browserstack by attaching this api on the browser console it worked fine

**Comment by BePo65** [comment]
Ok, I think I understand.

What this 'demopage' does is feature detection. for the fullscreen API. As webkitRequestFullscreen and webkitEnterFullscreen are features of an element, I test the availability for a `div` element based on code found in [MDN](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Feature_detection):

```javascript
"webkitRequestFullscreen" in  window.document.createElement('div');
"webkitEnterFullscreen" in  window.document.createElement('div');
```

Hope this helps.

**Comment by RopoMen** [comment]
For images I had an **idea which I haven't tested**. It's a quite a hack, but still. And this idea works only for putting a single image into fullscreen (not a carousel)

For Safari + IOS instead of using `img` tag try to use `video` tag with `poster` attribute that contains the image to be shown. Then use `webkitEnterFullscreen` for the video element.
